### PR TITLE
[FIX] point_of_sale: Update raspberry pi os for iot use

### PR DESCRIPTION
**Pull request by qle-odoo**
With this update the version of python goes to 3.9.2 In addition, a unique password system has been put in place

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qle-odoo** [comment]
@odoo/rd-security 
Using subprocess call to change PI user password in iot

File: addons/hw_drivers/tools/helpers.py [hunk 1]
```diff
@@ -188,6 +191,22 @@ def save_conf_server(url, token, db_uuid, enterprise_code):
...
     write_file('odoo-enterprise-code.conf', enterprise_code or '')
 
+def generate_password():
+    """
+    Generate an unique code to secure raspberry pi
+    """
+    alphabet = string.ascii_letters + string.digits
+    password = ''.join(secrets.choice(alphabet) for i in range(12))
+    salt = ''.join(secrets.choice(alphabet) for i in range(12))
+    shadow_password = crypt.crypt(password, '$1$' + salt + '$')
```
Comment (tbe-odoo) [inline addons/hw_drivers/tools/helpers.py:201]: Nothing particular to say regarding the code, by reading the documentation we can see that the salt generation can off-loaded to the crypt library by using `crypt.mksalt`

File: addons/hw_posbox_homepage/controllers/main.py [hunk 2]
```diff
@@ -253,6 +253,11 @@ def server(self):
             'loading_message': 'Configure Domain Server'
         })
 
+    # Get password
+    @http.route('/hw_posbox_homepage/password', type='json', auth='none', method=['POST'])
+    def view_password(self):
+        return helpers.generate_password()
```
[the remark this answers is not in the document]
Comment (qle-odoo) [reply addons/hw_posbox_homepage/controllers/main.py:259]: Yes, the password is displayed only once
after it is encrypted and if the user has not noted it, another one will have to be generated

File: addons/hw_drivers/tools/helpers.py [hunk 3]
```diff
@@ -188,6 +191,22 @@ def save_conf_server(url, token, db_uuid, enterprise_code):
...
+    subprocess.call(('sudo', 'usermod', '-p', shadow_password, 'pi'))
```
Comment (mart-e) [inline addons/hw_drivers/tools/helpers.py:202]: So if we call twice the route, we generate two different password?

File: addons/hw_drivers/tools/helpers.py [hunk 4]
```diff
@@ -188,6 +191,22 @@ def save_conf_server(url, token, db_uuid, enterprise_code):
     write_file('odoo-db-uuid.conf', db_uuid or '')
     write_file('odoo-enterprise-code.conf', enterprise_code or '')
 
+def generate_password():
+    """
+    Generate an unique code to secure raspberry pi
+    """
+    alphabet = string.ascii_letters + string.digits
```
Comment (mart-e) [inline addons/hw_drivers/tools/helpers.py:198]: maybe same as here and remove il1o0
https://github.com/odoo/odoo/blob/master/addons/web/static/src/public/database_manager.js#L38

**Review by mart-e — APPROVED** [review]

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: route protected via cross against silent malicious JS. subprocess not injectable. Would not work on a server without pi user.

**Comment by qle-odoo** [comment]
@robodoo r+

**Comment by lse-odoo** [comment]
@qle-odoo could we consider fwporting this PR in 15.0 ? I got a case where they are in 15 but as the image is pre-built with 17 there is a random password set with no way to regenerate :/ It's not the only case that I had with this issue even if it's pretty rare
opw-3957086

**Comment by qle-odoo** [comment]
> @qle-odoo could we consider fwporting this PR in 15.0 ? I got a case where they are in 15 but as the image is pre-built with 17 there is a random password set with no way to regenerate :/ It's not the only case that I had with this issue even if it's pretty rare opw-3957086

Yes and no
he can always retrieve a password before linking the iot.
when iot is still in 17.0

**Pull request closed without merging**